Migration page for Dart and Flutter v7 (#6090)
getsentry/sentry-docs: src/platforms/dart/migration.mdx

description: "Migrate between versions of Sentry's SDK for Dart."
---

## Migrating From `sentry` `6.18.x` to `sentry` `7.0.0`

API changes:

- Sentry's Dart SDK version 7.0.0 and above requires Dart `2.17.0`.
- Methods that used to take a `dynamic hint` optional parameter now takes `Hint? hint` instead.
- Remove deprecated fields from the `SentryDevice` class.
  - `screenResolution`, using the `screenHeightPixels` and `screenWidthPixels` instead.
  - `timezone`, using the `SentryCulture#timezone` instead.
  - `language`, using the `SentryCulture#locale` instead.
  - `theme`, using the `SentryOperatingSystem#theme` instead.
- Remove deprecated field from the `Contexts#dart_context` databag.

>   - `isolate`, using the `SentryThread#name` instead.

Author: Nope, `isolate` is the name of the field, so I'd say its similar to the suggestions above, using the "replaced with" wording.

>   - `isolate`, using the `SentryThread#name` instead.

Author: [`35de412` (#6090)](https://github.com/getsentry/sentry-docs/pull/6090/commits/35de412136594524f3478a2d4824e229b1585e94)

> - `SentryMeasurementUnit` has been redesigned to be strongly typed.
>   - `DurationSentryMeasurementUnit`
>   - `InformationSentryMeasurementUnit`
>   - `FractionSentryMeasurementUnit`
>   - `CustomSentryMeasurementUnit`
>   - `NoneSentryMeasurementUnit`

Reviewer: ```suggestion
-  The following`SentryMeasurementUnit`s are now strongly typed:

  - `DurationSentryMeasurementUnit`
  - `InformationSentryMeasurementUnit`
  - `FractionSentryMeasurementUnit`
  - `CustomSentryMeasurementUnit`
  - `NoneSentryMeasurementUnit`
```

> - Remove deprecated fields from the `Scope` class.
>   - `user(SentryUser? user)`, using the `setUser(SentryUser? user)` instead.
>   - `attachements`, using the `attachments` instead.

Reviewer: ```suggestion
- The following fields have been removed from the `Scope` class and replaced:

  - `user(SentryUser? user)` replaced with `setUser(SentryUser? user)`.
  - `attachements` replaced with `attachments`.
```

> - Classes or methods that used to take the bellow optional parameters, have been moved to the `SentryOptions` class:
>   - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests` instead.
>   - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii` instead.
>   - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize` instead.
>   - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
>   - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs` instead.

Reviewer: ```suggestion
- Classes or methods that used to take the below optional parameters, have been moved to the `SentryOptions` class and replaced:
  - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests`.
  - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii`.
  - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize`.
  - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler`.
  - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs`.
```

> - Classes or methods that used to take the bellow optional parameters were moved to the `SentryOptions` class.
>   - `captureFailedRequests`, using `SentryOptions#captureFailedRequests` instead.
>   - `sendDefaultPii`, using `SentryOptions#sendDefaultPii` instead.
>   - `maxRequestBodySize`, using `SentryOptions#maxRequestBodySize` instead.
>   - `networkTracing`, using `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
>   - `recordBreadcrumbs`, using `SentryOptions#recordHttpBreadcrumbs` instead.

Reviewer: ```suggestion
- Classes or methods that used to take the bellow optional parameters, have been moved to the `SentryOptions` class:

  - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests` instead.
  - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii` instead.
  - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize` instead.
  - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
  - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs` instead.
```

> -  The following`SentryMeasurementUnit`s are now strongly typed:
>   - `DurationSentryMeasurementUnit`
>   - `InformationSentryMeasurementUnit`
>   - `FractionSentryMeasurementUnit`
>   - `CustomSentryMeasurementUnit`
>   - `NoneSentryMeasurementUnit`

Reviewer: ```suggestion
-  The following `SentryMeasurementUnits` are now strongly typed:
   -  `DurationSentryMeasurementUnit`
   -  `InformationSentryMeasurementUnit`
   -  `FractionSentryMeasurementUnit`
   -  `CustomSentryMeasurementUnit`
   - ` NoneSentryMeasurementUnit`
```
Want to make sure these are indented the same way the other ones are :)

description: "Migrate between versions of Sentry's SDK for Dart."
---

## Migrating From `sentry` `6.18.x` to `sentry` `7.0.0`

API changes:

- Sentry's Dart SDK version 7.0.0 and above requires Dart `2.17.0`.
- Methods that used to take a `dynamic hint` optional parameter now take `Hint? hint` instead.
- The following deprecated fields have been removed from the `SentryDevice` class and replaced:

  - `screenResolution` replaced with `screenHeightPixels` and `screenWidthPixels`.
  - `timezone` replaced with `SentryCulture#timezone`.
  - `language` replaced with `SentryCulture#locale`.
  - `theme` replaced with `SentryOperatingSystem#theme`.

> - Remove deprecated field from the `Contexts#dart_context` databag.
>   - `isolate`, using the `SentryThread#name` instead.

Author: ```suggestion
- The following deprecated field has been removed from the `Contexts#dart_context` databag:
  - `isolate` replaced with `SentryThread#name`.
```
